test/parallel/test-fs-watch-recursive-promise.js

```diff
@@ -0,0 +1,60 @@
+'use strict';
+
+const common = require('../common');
+
+if (!common.hasCrypto)
+  common.skip('missing crypto');
+
+if (common.isIBMi)
+  common.skip('IBMi does not support `fs.watch()`');
+
+// fs-watch on folders have limited capability in AIX.
+// The testcase makes use of folder watching, and causes
+// hang. This behavior is documented. Skip this for AIX.
+
+if (common.isAIX)
+  common.skip('folder watch capability is limited in AIX.');
+
+const { randomUUID } = require('crypto');
```

Comment:
If you're only using the `crypto` module to generate UUID based filenames, then it might be worth doing that a different way (with a counter for example). That would let us drop the `common.hasCrypto` checks and run the tests with more configurations.